### ISSUE TITLE
SIGSEGV using FT.SEARCH with highlight

### ISSUE THREAD

**Issue by michael-grunder** [RedisLabsModules/RediSearch#312]
Hi, 

Thanks for writing RediSearch, it is really great!

I'm seeing a SIGSEGV when searching an index that should be pretty easy to reproduce.  I was able to cause segfault on OSX and Ubuntu by doing the following:

Create an index and add a document:
```bash
"FT.CREATE" "netflix" "SCHEMA" "title" "TEXT" "WEIGHT" "1" "rating" "TEXT" "WEIGHT" "1" "level" "TEXT" "WEIGHT" "1" "description" "TEXT" "WEIGHT" "1" "year" "NUMERIC" "uscore" "NUMERIC" "usize" "NUMERIC"
"FT.ADD" "netflix" "15ad80086ccc7f" "1" "FIELDS" "title" "The Vampire Diaries" "rating" "TV-14" "level" "Parents strongly cautioned. May be unsuitable for children ages 14 and under." "description" "90" "year" "2017" "uscore" "91" "usize" "80"
```

Now search with a highlight:
```bash
127.0.0.1:9999> ft.search netflix vampire highlight
```

Crashes in `FragmentList_HighlightWholeDocS`on the `memcpy`:
```bash
==29194== Thread 7 thread-pool-2:
==29194== Invalid write of size 1
==29194==    at 0x4C3678B: memmove (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29194==    by 0x7D2C313: memcpy (string_fortified.h:34)
==29194==    by 0x7D2C313: FragmentList_HighlightWholeDocS (fragmenter.c:231)
==29194==    by 0x7D2E5D3: summarizeField (highlight_processor.c:164)
==29194==    by 0x7D2E5D3: processField.isra.3 (highlight_processor.c:236)
==29194==    by 0x7D2E95A: hlp_Next (highlight_processor.c:292)
==29194==    by 0x7D45D22: ResultProcessor_Next (result_processor.c:48)
==29194==    by 0x7D439DC: Query_SerializeResults (query_plan.c:91)
==29194==    by 0x7D4405F: QueryPlan_Run (query_plan.c:215)
==29194==    by 0x7D36265: _SearchCommand (module.c:928)
==29194==    by 0x7D25CFE: threadHandleCommand (concurrent_ctx.c:38)
==29194==    by 0x7D26A13: thread_do (thpool.c:337)
==29194==    by 0x539D7FB: start_thread (pthread_create.c:465)
==29194==    by 0x56C9B5E: clone (clone.S:95)
==29194==  Address 0x5e616ca is 0 bytes after a block of size 10 alloc'd
==29194==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29194==    by 0x7D2C2E6: FragmentList_HighlightWholeDocS (fragmenter.c:225)
==29194==    by 0x7D2E5D3: summarizeField (highlight_processor.c:164)
==29194==    by 0x7D2E5D3: processField.isra.3 (highlight_processor.c:236)
==29194==    by 0x7D2E95A: hlp_Next (highlight_processor.c:292)
==29194==    by 0x7D45D22: ResultProcessor_Next (result_processor.c:48)
==29194==    by 0x7D439DC: Query_SerializeResults (query_plan.c:91)
==29194==    by 0x7D4405F: QueryPlan_Run (query_plan.c:215)
==29194==    by 0x7D36265: _SearchCommand (module.c:928)
==29194==    by 0x7D25CFE: threadHandleCommand (concurrent_ctx.c:38)
==29194==    by 0x7D26A13: thread_do (thpool.c:337)
==29194==    by 0x539D7FB: start_thread (pthread_create.c:465)
==29194==    by 0x56C9B5E: clone (clone.S:95)
```

It looks like a possible underflow/overflow but I could be wrong:
```C
for (size_t ii = 0; ii < niovs; ++ii) {                                                                                                           
	// GDB: ii=3, niovs=4
    // GDB: iovs[ii] = {iov_base = 0x7fffe2c1b077, iov_len = 18446744073709551614}
    memcpy(docBuf + offset, iovs[ii].iov_base, iovs[ii].iov_len);                                                                                   
    offset += iovs[ii].iov_len;
}
```

I'm running RediSearch `master` (f6fa1cb9) and redis-server 4.0.9 (8ac7af1c) and can replicate this on OSX and Ubuntu.

Hopefully, it's not too hard to track down!

Cheers,
Mike

**Comment by dvirsky** [comment]
Wow, what a great bug report! :)

**Comment by dvirsky** [comment]
@mnunberg any news on this?

**Comment by mnunberg** [comment]
Fixed in 1.0

**Comment by michael-grunder** [comment]
Wow, thanks for the quick fix!  This is now working perfectly on my end. 🙂
